src/vm/jvm/runtime/org/raku/nqp/sixmodel/reprs/VMArray.java

```diff
@@ -88,7 +88,43 @@ public void compose(ThreadContext tc, STable st, SixModelObject repr_info) {
     public SixModelObject deserialize_stub(ThreadContext tc, STable st) {
         SixModelObject obj;
         if (st.REPRData == null) {
-            obj = new VMArrayInstance();
+            // Either a real VMArray or REPRData not yet known.
+            switch (st.REPR.name) {
+            case "VMArray":
+                obj = new VMArrayInstance();
+                break;
+            case "VMArray_i8":
+                obj = new VMArrayInstance_i8();
+                break;
+            case "VMArray_u8":
+                obj = new VMArrayInstance_u8();
+                break;
+            case "VMArray_i16":
+                obj = new VMArrayInstance_i32();
```

Comment:
I only now noticed, that this should be ```VMArrayInstance_i16()``` (same copy-pasto for the unsinged variant). I'll force push a correction.